src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutesV2Spec.scala

```diff
@@ -0,0 +1,406 @@
+        .thenReturn(IO(true)))
+      missingActionsOnNewParent.map(action => when(samRoutes.policyEvaluatorService.hasPermission(mockitoEq(newParent), mockitoEq(action), mockitoEq(defaultUserInfo.userId), any[SamRequestContext]))
+        .thenReturn(IO(false)))
+      if (accessToNewParent) {
+        when(samRoutes.policyEvaluatorService.listResourceAccessPoliciesForUser(mockitoEq(newParent), mockitoEq(defaultUserInfo.userId), any[SamRequestContext]))
+          .thenReturn(IO(Set(otherPolicy)))
+      } else {
+        when(samRoutes.policyEvaluatorService.listResourceAccessPoliciesForUser(mockitoEq(newParent), mockitoEq(defaultUserInfo.userId), any[SamRequestContext]))
+          .thenReturn(IO(Set[AccessPolicyWithoutMembers]()))
+      }
+    }
+  }
+
+  "GET /api/resources/v2/{resourceTypeName}/{resourceId}/parent" should "200 if user has get_parent on resource and resource has parent" in {
+    val fullyQualifiedChildResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("child"))
+    val fullyQualifiedParentResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("parent"))
+    val samRoutes = createSamRoutes()
+
+    setupRoutesTest(samRoutes, fullyQualifiedChildResource,
+      currentParentOpt = Option(fullyQualifiedParentResource),
+      actionsOnChild = Set(SamResourceActions.getParent))
+
+    Get(s"/api/resources/v2/${defaultResourceType.name}/${fullyQualifiedChildResource.resourceId.value}/parent") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[FullyQualifiedResourceId] shouldEqual fullyQualifiedParentResource
+    }
+  }
+
+  it should "403 if user is missing get_parent on resource" in {
+    val fullyQualifiedChildResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("child"))
+    val fullyQualifiedParentResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("parent"))
+    val samRoutes = createSamRoutes()
+
+    setupRoutesTest(samRoutes, fullyQualifiedChildResource, currentParentOpt = Option(fullyQualifiedParentResource),
+      missingActionsOnChild = Set(SamResourceActions.getParent))
+
+    Get(s"/api/resources/v2/${defaultResourceType.name}/${fullyQualifiedChildResource.resourceId.value}/parent") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+    }
+  }
+
+  it should "404 if resource has no parent" in {
+    val fullyQualifiedChildResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("child"))
+    val samRoutes = createSamRoutes()
+
+    setupRoutesTest(samRoutes, fullyQualifiedChildResource, None, actionsOnChild = Set(SamResourceActions.getParent))
+
+    Get(s"/api/resources/v2/${defaultResourceType.name}/${fullyQualifiedChildResource.resourceId.value}/parent") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.NotFound
+    }
+  }
+
+  // this one feels a bit like it's testing the security directives not the route itself but whatever
```

Comment:
think of Route specs as contract tests... we need to make sure that we return responses with the right codes and format

Reply:
@dvoet so do you think I should change the tests to not worry about missing specific permissions on a resource and instead to just test that when some permission is missing it returns a 403?